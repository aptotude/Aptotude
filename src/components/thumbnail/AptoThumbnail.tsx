import React from 'react';
import classNames from 'classnames';
import { StandardTypes } from '../../utils/standardTypes';
import './aptoThumbnail.scss';

const COMPONENT_PREFIX = 'AptoThumbnail';

export class AptoThumbnail extends React.Component<StandardTypes> {
  public render() {
    const {
      className,
      forwardRef,
      children,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      className
    );

    return (
      <div
        {...rest}
        ref={forwardRef}
        className={classes}>
          {children}
      </div>
    );
  }
}
