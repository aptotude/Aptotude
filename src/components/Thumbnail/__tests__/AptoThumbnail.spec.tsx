import React from 'react';
import renderer from 'react-test-renderer';
import AptoThumbnail from '../AptoThumbnail';

test('Thumbnail renders', () => {
  const component = renderer.create(
    <AptoThumbnail />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Thumbnail with image renders', () => {
  const component = renderer.create(
    <AptoThumbnail>
        <img src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==" />
    </AptoThumbnail>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
