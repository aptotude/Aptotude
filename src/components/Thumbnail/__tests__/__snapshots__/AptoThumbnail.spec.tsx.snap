// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail renders 1`] = `
<div
  className="AptoThumbnail"
/>
`;

exports[`Thumbnail with image renders 1`] = `
<div
  className="AptoThumbnail"
>
  <img
    src="data:image/png;base64, iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
  />
</div>
`;
